File: Unit4E3.ts
/**
 * @author William
 * @version 1.0.0
 * @date 2025-11-27
 * @fileoverview This program calculates the sum of even numbers up to a positive integer.
 */

// variables
let n: number = 0;
let sum: number = 0;

// ask for positive integer
do {
  n = parseInt(prompt("Enter a positive integer: ") || "0");

  if (n <= 0) {
    console.log("Invalid input. Please enter a positive integer:");
  }
} while (n <= 0);

// calculate sum of even numbers
for (let counter = 1; counter <= n; counter++) {
  if (counter % 2 === 0) {
    sum = sum + counter;
  }
}

console.log(`Sum of even numbers from 1 to ${n} is: ${sum}`);

console.log("Done.");

File: Unit4E3.go
// Author: William
// Version: 1.0.0
// Date: 2025-11-27
// Fileoverview: This program calculates the sum of even numbers up to a user-entered positive integer.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	var n int
	var sum int = 0

	reader := bufio.NewReader(os.Stdin)

	// ask for positive integer
	for {
		fmt.Print("Enter a positive integer: ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		n, _ = strconv.Atoi(input)

		if n > 0 {
			break
		} else {
			fmt.Println("Invalid input. Please enter a positive integer:")
		}
	}

	// calculate sum of even numbers
	for counter := 1; counter <= n; counter++ {
		if counter%2 == 0 {
			sum += counter
		}
	}

	fmt.Printf("Sum of even numbers from 1 to %d is: %d\n", n, sum)
	fmt.Println("Done.")
}
